vyperlang/vyper
# chore: support vyper files for tests

### What I did

- Update `tests/conftest.py` to handle the testing of compilation outcomes for `.vy` files (as per https://docs.pytest.org/en/6.2.x/example/nonpython.html)

### How I did it

The Vyper files need to adhere to the syntax `test_DESCRIPTION_OUTCOME` where `DESCRIPTION` is non-consequential and `OUTCOME` is either `Success`, a `VyperException` class name, or `Fail` (unhandled exception). `OUTCOME` is extracted upon the initialisation of each Vyper test file.

For each test file, we attempt to compile it using `compile_files`, and check the compilation outcome against the extracted `OUTCOME`.

Example of a failing test:
![Screenshot from 2022-02-26 17-26-40](https://user-images.githubusercontent.com/8017125/155837887-5ac13f36-70e7-4836-8e78-eb99d656c85e.png)


### How to verify it

Run `pytest tests/parser/syntax/constants/` to see a demo against two sample contracts from `test_constants.py`. You can also edit the Vyper test files to see what a failing test look like.

### Description for the changelog

Handle Vyper test files

### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://i.pinimg.com/736x/0a/bb/a4/0abba415ba07ca88847658216c1c537f.jpg)
